sdk/communication/communication-call-automation/src/utli/parser.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createIdentifierFromRawId } from "@azure/communication-common";
+import {
+  TranscriptionDataInternal,
+  TranscriptionMetadata,
+  TranscriptionData,
+} from "../models/transcription";
+
+/** Parse the incoming package. */
+export function streamingDataParser(
+  packetData: string | BinaryData,
+): TranscriptionMetadata | TranscriptionData {
+  let stringJson: string;
+  if (typeof packetData === "string") {
+    stringJson = packetData;
+  } else {
+    const decoder = new TextDecoder();
+    stringJson = decoder.decode(packetData);
+  }
+
+  const jsonObject = JSON.parse(stringJson);
+  const kind: string = jsonObject.kind;
+
+  switch (kind) {
+    case "TranscriptionMetadata": {
+      const transcriptionMetadata: TranscriptionMetadata = jsonObject.transcriptionMetadata;
+      return transcriptionMetadata;
+    }
+    case "TranscriptionData": {
+      const transcriptionDataInternal: TranscriptionDataInternal = jsonObject.transcriptionData;
```

Comment:
Why do we need to have this middle step?

Reply:
removed it.